gbpservice/nfp/config/mode_shift.sh

```diff
@@ -0,0 +1,164 @@
+    VISIBILITY_QCOW2_IMAGE_NAME=visibility
+    InstanceName="VisibilityVM_instance"
+    create_port_for_vm $VISIBILITY_QCOW2_IMAGE_NAME
+
+    if [[ $VISIBILITY_QCOW2_IMAGE = build ]]; then
+       # edits the docker file to add visibility vm IP address
+       configure_vis_ip_addr_in_docker
+
+       # prepare visibility source, this is needed for diskimage build
+       cd /home/stack/
+       sudo rm -rf visibility
+       sudo git clone https://$GIT_ACCESS_USERNAME:$GIT_ACCESS_PASSWORD@github.com/oneconvergence/visibility.git -b $VISIBILITY_GIT_BRANCH
+       echo "Building Image: $VISIBILITY_QCOW2_IMAGE_NAME"
+       cd $DEST/gbp/gbpservice/tests/contrib/diskimage-create/
+       sudo python visibility_disk_image_create.py visibility_conf.json $DEVSTACK_DIR/local.conf
+       VISIBILITY_QCOW2_IMAGE=$(cat /tmp/image_path)
+    fi
+    echo "Uploading Image: $VISIBILITY_QCOW2_IMAGE_NAME"
+    glance image-create --name $VISIBILITY_QCOW2_IMAGE_NAME --disk-format qcow2 --container-format bare --visibility public --file $VISIBILITY_QCOW2_IMAGE
+    sleep 4
+    
+    if ! [[ -z $AsavQcow2Image ]]; then
+        gbp service-profile-create --servicetype FIREWALL --insertion-mode l3 --shared True --service-flavor service_vendor=asav,device_type=nova --vendor NFP asav_fw_profile
+
+        ASAV_QCOW2_IMAGE_NAME=asav
+        echo "Uploading Image: $ASAV_QCOW2_IMAGE_NAME"
+        glance image-create --name $ASAV_QCOW2_IMAGE_NAME --disk-format qcow2 --container-format bare --visibility public --file $AsavQcow2Image
+    fi
+}
+
+function nfp_configure_nova {
+    NOVA_CONF_DIR=/etc/nova
+    NOVA_CONF=$NOVA_CONF_DIR/nova.conf
+    source $DEVSTACK_DIR/inc/ini-config
+    iniset $NOVA_CONF DEFAULT instance_usage_audit "True"
+    
+    source $DEVSTACK_DIR/functions-common
+    stop_process n-cpu
+    stop_process n-cond 
+    stop_process n-sch 
+    stop_process n-novnc 
+    stop_process n-cauth
+    stop_process n-api 
+    
+    source $DEVSTACK_DIR/lib/nova
+    start_nova_compute
+    start_nova_api
+    run_process n-cond "$NOVA_BIN_DIR/nova-conductor --config-file $NOVA_CONF"
+    run_process n-sch "$NOVA_BIN_DIR/nova-scheduler --config-file $NOVA_CONF"
+    run_process n-novnc "$NOVA_BIN_DIR/nova-novncproxy --config-file $NOVA_CONF --web $DEST/noVNC"
+    run_process n-cauth "$NOVA_BIN_DIR/nova-consoleauth --config-file $NOVA_CONF"
+}
+
+function prepare_for_mode_shift {
+    if [[ $FROM = advanced ]] && [[ $TO = enterprise ]]; then
+        source $DEST/gbp/devstack/lib/nfp
+
+        echo "Preparing image creation"
+        create_images
+
+        echo "Launching the Visibility VM"
+        launch_visibilityVM
+
+        nfp_logs_forword
+        nfp_configure_nova
+    else
+        echo "Shifting from $FROM mode to $TO mode is not supported."
+    fi
+}
+
+function delete_instance_and_image {
+    
+    # delete the instance
+    echo "Deleting the running '$2' instance."
+    nova delete $2
+    sleep 5
```

Comment:
Might not be enough. Ideally, we should check if VM got deleted using nova command. This might fail if the setup is slow.